scripts/screeps_tencent_batch_rl_runner.py

```
@@ -918,6 +971,387 @@ def remote_training_report_path(artifact_dir: Path, run_id: str) -> Path:
+    scenario = first_dict(reports, (("scenario",), ("experimentCard", "scenario")))
+    if scenario is not None and not scenario_is_e1s1_single_room_no_hostile(scenario):
+        return False
+    if scenario_id is None and scenario is None:
+        return False
+    room = first_text(reports, (("simulation", "room"), ("source", "initialConditions", "room")))
+    if room is not None and room != "E1S1":
+        return False
+    map_source_file = extract_map_source_file(reports)
+    if map_source_file is not None and Path(map_source_file).name != "map-0b6758af.json":
+        return False
+    return True
```

Ruff (0.15.12)

[warning] Return the negated condition directly

Inline condition

(SIM103)